### PR TITLE
SISRP-17446, explain why 403 on /advising/student page; clickable email

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.berkeley.edu/browse/SISRP-17446

User gets:
* error summary
* clickable email address 
* spinner stop via `finally`

### CODE REVIEW

File: src/assets/javascripts/angular/controllers/pages/advisingStudentController.js [hunk 1]
```diff
@@ -29,20 +45,26 @@ angular.module('calcentral.controllers').controller('AdvisingStudentController',
...
     });
   };
 
   var loadAcademics = function() {
     advisingFactory.getStudentAcademics({
       uid: $routeParams.uid
-    }).then(function(data) {
-      $scope.academics = _.get(data, 'data.academics');
+    }).success(function(data) {
+      $scope.academics = _.get(data, 'academics');
```
Comment (paulkerschen) [inline src/assets/javascripts/angular/controllers/pages/advisingStudentController.js]: @johncrossman Is `_get` needed here, now that the intervening `.data` object is gone?

Comment (johncrossman) [inline src/assets/javascripts/angular/controllers/pages/advisingStudentController.js]: @paulkerschen I had same question. We have similar usages in *enrollmentCardController.js*, *profilePhoneController.js*, etc. but I doubt lodash is needed here. I'll test.

**Comment by johncrossman** [comment]
Interesting lesson learned, @paulkerschen   I actually need `_.get()` in these cases but for an unexpected reason. I need a default of `{}` when *data.attributes* is nil because subsequent lines add elements to the hash. Amendment is in.

**Comment by johncrossman** [comment]
Lo and behold, scrutinizing the `_get()` led me to a related problem: `student.isLoading` status was lost when `$scope.student` was set. This has also been amended (see `studentProfile.isLoading`).  cc: @paulkerschen

**Comment by johncrossman** [comment]
@paulkerschen I'm going to use `angular.extend` instead.  Please wait for one more amendment.

File: src/assets/templates/widgets/advisor/student_profile.html [hunk 2]
```diff
@@ -2,82 +2,90 @@
...
-            <br/>
-            <div data-ng-repeat="phone in student.phones">
-              <div>
-                <strong data-ng-bind-template="{{phone.type.description}} Phone"></strong>
-                <span data-ng-if="phone.primary">(preferred)</span>
+              <br/>
+              <div data-ng-repeat="email in student.emails">
+                <div>
+                  <strong data-ng-bind-template="{{email.type.description}} Email"></strong>
+                  <span data-ng-if="email.primary">(preferred)</span>
```
Comment (paulkerschen) [inline src/assets/templates/widgets/advisor/student_profile.html]: @johncrossman I see this was in the existing code, but just checking: on email and phone, "primary" is synonymous with "preferred"? I believe they mean different things when translating back-end CS feeds.

Comment (johncrossman) [inline src/assets/templates/widgets/advisor/student_profile.html]: @paulkerschen  Yes, w.r.t email and phone, 'primary' is synonymous with 'preferred'. It's Hub naming translated for CalCentral user experience. The confusion continues with the Hub's "names" schema in which a name item has both:
* `type.code = [PRI | PRF]`
* `preferred: [true | false]`

Comment (paulkerschen) [inline src/assets/templates/widgets/advisor/student_profile.html]: Thanks, @johncrossman, it's that name schema I was thinking of.

**Pull request merged**